internal/core/sharedvector.rs

```diff
@@ -315,19 +315,12 @@ impl<T: Clone> From<&[T]> for SharedVector<T> {
     }
 }
 
-macro_rules! from_array {
-    ($($n:literal)*) => { $(
-        // FIXME: remove the Clone bound
-        impl<T: Clone> From<[T; $n]> for SharedVector<T> {
-            fn from(array: [T; $n]) -> Self {
-                array.iter().cloned().collect()
-            }
-        }
-    )+ };
+impl<T: Clone, const N: usize> From<[T; N]> for SharedVector<T> {
```

Comment:
Ah, the Clone bound is still needed?

Reply:
No, this is a leftover

```suggestion
impl<T, const N: usize> From<[T; N]> for SharedVector<T> {
```